cluster-codec-parent/cluster-codec-jackson/src/main/java/io/scalecube/cluster/codec/jackson/JacksonMetadataEncoder.java

```diff
@@ -0,0 +1,72 @@
+package io.scalecube.cluster.codec.jackson;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.scalecube.cluster.metadata.MetadataDecoder;
+import io.scalecube.cluster.metadata.MetadataEncoder;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Map;
+import reactor.core.Exceptions;
+
+/**
+ * Jackson-based metadata encoder/decoder.
+ *
+ * @author eutkin
+ */
+public class JacksonMetadataEncoder implements MetadataEncoder, MetadataDecoder {
+
+  private static final TypeReference<Map<String, String>> TYPE =
```

Comment:
`<String, String>` is too narrow typing. We want `Type` in general or `Class<?>`. Looks like need to add new method into decoder of metadata interface. Hold on, I will create separate PR branched from this one.

Reply:
@artem-v WDYT about https://github.com/scalecube/scalecube-cluster/pull/287#discussion_r363041187
Adding `Type` and `Class<?>` to arguments will bring more redundant complexity, because custom implementations will have to deal with this type, check it out, adjust to it, etc. And IMHO lots of custom implementations of the interface will just ignore this parameter.
All encoding tools can persist enough type information to deserialize the target in the future. For instance: `sbe` has templateId, `protostuff` has schema and `jackson` can do it via type id - https://github.com/FasterXML/jackson-docs/wiki/JacksonPolymorphicDeserialization#instance-type-information-type-metadata

Reply:
50% of users will not use metadata.
40% of users will use <String, String> as metadata.
10% will consider to customize metadata.

we need to allow it to be customized but not create complexity

Reply:
All your concerns has been addressed in 
https://github.com/scalecube/scalecube-cluster/pull/289

Reply:
@ronenhamias 
On seed node metadata is null, on other nodes u have metadata. <String, String>` didn't work for us, it's good for pet project, but not for something more or less real.

Reply:
@segabriel 
Regarding `sbe` true, regarding `jackson` true, regarding `protostuff` not true, you have to have type/class. Regarding gson .. again, not true as far as I researched.

Reply:
> @ronenhamias
> On seed node metadata is null, on other nodes u have metadata. <String, String>` didn't work for us, it's good for pet project, but not for something more or less real.

so in that case this ability to change metadata decoders is a risky feature.
as cluster decoders might be inconsistent and error prone.
i think its better to have one codec metadata <String,bytes>
and user can handle the bytes as he want.